packages/arb-validator-core/gobridge/arbRollup.go

```diff
@@ -543,22 +534,90 @@ func (vm *arbRollup) Confirm(ctx context.Context, opp *valprotocol.ConfirmOpport
 
 	confirmedEvent := arbbridge.ConfirmedEvent{
 		ChainInfo: arbbridge.ChainInfo{
-			BlockId: vm.Client.getCurrentBlock(),
+			BlockId: vm.client.getCurrentBlock(),
 		},
 		NodeHash: confNode,
 	}
-	vm.Client.pubMsg(nil, arbbridge.MaybeEvent{
+	vm.client.pubMsg(nil, arbbridge.MaybeEvent{
 		Event: confirmedEvent,
 	})
 
-	confirmedAssertionEvent := arbbridge.ConfirmedAssertionEvent{
-		ChainInfo: arbbridge.ChainInfo{
-			BlockId: vm.Client.getCurrentBlock(),
-		},
+	// Send all messages is a single batch
+	//globalInbox.sendMessages(data.messages);
+	for _, msg := range messages {
+		//sendMsg
+		msgTypeVal, _ := msg.(value.TupleValue).GetByInt64(0)
+		msgTypeInt, ok := msgTypeVal.(value.IntValue)
+		if !ok {
+			return errors.New("msg type must be an int")
+		}
+		switch message.MessageType(msgTypeInt.BigInt().Uint64()) {
+		case message.EthType:
+			ethMsg, err := message.UnmarshalEth(msg)
+			if err != nil {
+				return errors.New("invalid message")
+			}
+			err = transferEth(vm.client.goEthdata, ethMsg)
+		case message.ERC20Type:
+			//erc20Msg, err := message.UnmarshalERC20(msg)
+			//if err != nil {
+			//	return errors.New("invalid message")
+			//}
+			//err = transferEth(vm.client.goEthdata, ethMsg)
+		case message.ERC721Type:
+
+		}
```

Comment:
Looks like some work to be finished here.

Reply:
fixed